wicker/core/config.py

```diff
@@ -23,33 +23,65 @@ def from_json(cls, data: Dict[str, Any]) -> WickerWandBConfig:
         )
 
 
+@dataclasses.dataclass(frozen=True)
+class BotoS3Config:
+    max_pool_connections: int
+    read_timeout: int
+    connect_timeout: int
+
+    @classmethod
+    def from_json(cls, data: Dict[str, Any]) -> BotoS3Config:
+        return cls(
+            max_pool_connections=data["max_pool_connections"],
+            read_timeout=data["read_timeout"],
+            connect_timeout=data["connect_timeout"],
+        )
+
+
 @dataclasses.dataclass(frozen=True)
 class WickerAwsS3Config:
     s3_datasets_path: str
     region: str
+    boto_config: BotoS3Config
     store_concatenated_bytes_files_in_dataset: bool = False
 
     @classmethod
     def from_json(cls, data: Dict[str, Any]) -> WickerAwsS3Config:
         return cls(
             s3_datasets_path=data["s3_datasets_path"],
             region=data["region"],
+            boto_config=BotoS3Config.from_json(data["boto_config"]),
             store_concatenated_bytes_files_in_dataset=data.get("store_concatenated_bytes_files_in_dataset", False),
         )
 
 
+@dataclasses.dataclass(frozen=True)
+class S3StorageConfig:
+    retries: int
+    timeout: int
+
+    @classmethod
+    def from_json(cls, data: Dict[str, Any]) -> S3StorageConfig:
+        return cls(
+            retries=data["retries"],
+            timeout=data["timeout"],
+        )
+
+
 @dataclasses.dataclass(frozen=True)
 class WickerConfig:
```

Comment:
Do we need to add the boto config to this object too? I think we should probably have it

Reply:
Sorry missed it in the config above, ignore this comment, you put it in a good place